go/vt/vtgate/engine/update.go

```diff
@@ -137,6 +140,59 @@ func (upd *Update) execUpdateEqual(vcursor VCursor, bindVars map[string]*querypb
 	return execShard(vcursor, upd.Query, bindVars, rs, true /* rollbackOnError */, true /* canAutocommit */)
 }
 
+func (upd *Update) execUpdateIn(vcursor VCursor, bindVars map[string]*querypb.BindVariable) (*sqltypes.Result, error) {
```

Comment:
I'm a little concerned about the amount of shared code with `execDeleteIn` in `delete.go`. If I understand it correctly, the only difference between the two is the `updateVindexEntries` vs `deleteVindexEntries`. Isn't there a way of avoiding this much copy-pasta?

Reply:
moved most of the part into function calls.